config/default/tasks.js

```diff
@@ -0,0 +1,334 @@
+        //label: "Delivery",
+        modifyContent: function (content, contact, report) {
+          content.pregnancy_uuid = report._id;
+        }
+      }
+    ],
+    events: [
+      {
+        id: "delivery-reminder",
+        start: 28, //due - 4 weeks 
+        end: 42, //due + 6 weeks
+        dueDate: function (event, contact, report) {
+          return addDays(getMostRecentLMPDateForPregnancy(contact, report), MAX_DAYS_IN_PREGNANCY); //LMP + 42 weeks
+        }
+      }
+    ]
+  },
+
+  {
+    icon: "icon-follow-up",
+    title: "task.pnc.danger_sign_followup_mother.title",
+    appliesTo: "reports",
+    appliesToType: ["delivery", "pnc_danger_sign_follow_up_mother"],
+    appliesIf: function (contact, report) {
+      return getField(report, 't_danger_signs_referral_follow_up') === 'yes' && isAlive(contact);
+    },
+    resolvedIf: function (contact, report, event, dueDate) {
+      //(refused or migrated) and clear tasks 
+      if (isPregnancyTaskMuted(contact)) return true;
+      const startTime = Math.max(addDays(dueDate, -event.start).getTime(), report.reported_date + 1);//+1 so that source ds_follow_up does not resolve itself;
+      const endTime = addDays(dueDate, event.end + 1).getTime();
+      return isFormArraySubmittedInWindow(contact.reports, ['pnc_danger_sign_follow_up_mother'], startTime, endTime);
+    },
+    actions: [
+      {
+        type: "report",
+        form: "pnc_danger_sign_follow_up_mother",
+        //label: "Pregnancy follow up",
+        modifyContent: function (content, contact, report) {
+          content.pregnancy_uuid = report._id;
+        }
+      }
+    ],
+    events: [
+      {
+        id: "pnc-danger-sign-follow-up-mother",
+        start: 3,
+        end: 7,
+        dueDate: function (event, contact, report) {
+          return getDateISOLocal(getField(report, 't_danger_signs_referral_follow_up_date'));
+        }
+      }
+    ]
+  },
+
+  {
+    icon: "icon-follow-up",
+    title: "task.pnc.danger_sign_followup_baby.title",
+    appliesTo: "contacts",
+    appliesToType: ["person"],
+    appliesIf: function (contact) {
+      return contact.contact &&
+        contact.contact.t_danger_signs_referral_follow_up === 'yes' &&
+        isAlive(contact);
+    },
+    resolvedIf: function (contact, report, event, dueDate) {
+      const startTime = Math.max(addDays(dueDate, -event.start).getTime(), contact.contact.reported_date);
+      const endTime = addDays(dueDate, event.end).getTime();
+      return isFormArraySubmittedInWindow(contact.reports, ['pnc_danger_sign_follow_up_baby'], startTime, endTime);
+    },
+    actions: [
+      {
+        type: "report",
+        form: "pnc_danger_sign_follow_up_baby"
+      }
+    ],
+    events: [
+      {
+        id: "pnc-danger-sign-follow-up-baby",
+        start: 3,
+        end: 7,
+        dueDate: function (event, contact) {
+          return getDateISOLocal(contact.contact.t_danger_signs_referral_follow_up_date);
```

Comment:
Is this right? The `t_danger_signs_referral_follow_up_date` attribute is directly on the contact?

Reply:
Yes, it was intended, This is for newborn baby contacts that are created through and marked as having danger signs in the delivery form.